packages/components/src/text-highlight/index.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { escapeRegExp } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { TextHighlightProps } from './types';
+
+/**
+ * Highlights occurrences of a given string within another string of text. Wraps
+ * each match with a `<mark>` tag which provides browser default styling.
+ *
+ * ```jsx
+ * import { TextHighlight } from '@wordpress/components';
+ *
+ * const MyTextHighlight = () => (
+ *   <TextHighlight
+ *     text="Why do we like Gutenberg? Because Gutenberg is the best!"
+ *     highlight="Gutenberg"
+ *   />
+ * );
+ * ```
+ */
+export const TextHighlight = ( props: TextHighlightProps ) => {
+	const { text = '', highlight = '' } = props;
```

Comment:
I think it makes sence to have them both marked as required. Not sure why you want to use this component if not both is provided?

Not sure I want to change the behavor in this PR. Trying to not change the behavor when refactoring to TypeScript. Maybe do a follow-up PR and it can be discussed more there?

Have added the `@default` notation.

Reply:
> Not sure I want to change the behavor in this PR. Trying to not change the behavor when refactoring to TypeScript.

Good point, sounds sensible